common/src/main/scala/explore/common/ConstraintSetObsQueries.scala

```diff
@@ -197,14 +197,9 @@ object ConstraintSetObsQueries {
   implicit val constraintSetWithObsReusability: Reusability[ConstraintSetsWithObs] =
     Reusability.derive
 
-  type LiveQueryRenderer =
-    (
-      View[ConstraintSetsWithObs] => VdomNode
-    ) => LiveQueryRenderMod[ObservationDB, ConstraintSetsObsQuery.Data, ConstraintSetsWithObs]
-
-  val ConstraintSetObsLiveQuery: LiveQueryRenderer =
-    render =>
-      AppCtx.runWithCtx { implicit appCtx =>
+  val ConstraintSetObsLiveQuery =
+    ScalaFnComponent[View[ConstraintSetsWithObs] => VdomNode](render =>
+      AppCtx.using { implicit appCtx =>
```

Comment:
Is it so that we execute this on `IO` when calling `using`? Or is it different than `runWitCtx` in that sense

Reply:
The inner workings are different than before. `Ctx.using` (or `Ctx.usingView`) return a `VdomElement`, so they can only be used within `render`. In other words, they act as a component; and since it's declarative, a name with `run` in it didn't seem fitting anymore. Also, the function passed to them must be a render function, returning a `VdomNode`.

This is a functionality provided by React, and is basically a way of "teleporting" props further down the component tree, without the need of explicitly passing them in all intermediate components. If the context changes, it is automatically propagated to all `.using(View)` blocks in the currently rendered component tree.